extensionsdk/Microsoft.Windows.DevHome.SDK/Microsoft.Windows.DevHome.SDK.idl

```diff
@@ -72,6 +72,18 @@ namespace Microsoft.Windows.DevHome.SDK
         Windows.Foundation.IAsyncOperation<ProviderOperationResult> CloneRepositoryAsync(IRepository repository, String cloneDestination, IDeveloperId developerId);
     };
 
+    [contract(Microsoft.Windows.DevHome.SDK.DevHomeContract, 2)] interface IRepositoryProvider2
+    requires IRepositoryProvider
+    {
+        String[] SearchFieldNames
+        {
+            get;
+        };
+
+        Windows.Foundation.IAsyncOperation<IVectorView<String> > GetValuesForSearchFieldAsync(String requestedSearchField, IMapView<String, String> fieldValues, IDeveloperId developerId);
+        Windows.Foundation.IAsyncOperation<RepositoriesResult> GetRepositoriesAsync(IMapView<String, String> fieldValues, IDeveloperId developerId);
```

Comment:
I would think that the parameters here should be switched, so that the first one matches the version of this method with only one parameter.

Reply:
I like that idea.  Changed.